lib/matplotlib/legend.py

```diff
@@ -506,11 +511,18 @@ def __init__(self, parent, handles, labels,
         self._set_loc(loc)
         self._loc_used_default = tmp  # ignore changes done by _set_loc
 
-        # figure out title fontsize:
-        if title_fontsize is None:
-            title_fontsize = mpl.rcParams['legend.title_fontsize']
-        tprop = FontProperties(size=title_fontsize)
-        self.set_title(title, prop=tprop)
+        # figure out title font properties:
+        title_prop_fp = FontProperties._from_any(title_fontproperties)
+        if isinstance(title_fontproperties, dict) and \
+           "size" not in title_fontproperties:
+            if title_fontsize is None:
+                title_fontsize = mpl.rcParams["legend.title_fontsize"]
+            title_prop_fp.set_size(title_fontsize)
+        else:
+            if title_fontsize is not None:
+                title_prop_fp.set_size(title_fontsize)
```

Comment:
I spent several minutes trying to figure what the precedence rules are and if they are reasonable.

I propose to forbid using both `title_fontsize` and `title_fontproperties` together. The combinations are awkward anyway: Specifying size in both is unclear; specifying size explicitly but e.g. family via `title_fontproperties` is clumsy - one should specify both via `title_fontproperties` in such a case.

The behavior should also be mentioned in the docstring.

Reply:
Or should we just deprecate `title_fontsize`, as suggested by @QuLogic above?

Reply:
@timhoffm?

Even if we deprecate though, we'd probably need to decide on precedence in the interim.

Reply:
I wouldn't not to deprecate `title_fontsize` for now. It's been in since 3.0 and forcing all users to switch is not justified. Also I'm slightly hesitant because `title_fontproperties={'size': 12}` is much more to write compared to `title_fontsize=12`.

If we forbid using both ways simultaneously, we don't have to decide on precedence.

Reply:
OK, thanks for your confirmation.